pkg/streamingpromql/operators/aggregations/common.go

```diff
@@ -35,3 +37,43 @@ var AggregationGroupFactories = map[parser.ItemType]AggregationGroupFactory{
 //
 // Invalid combinations include exponential and custom buckets, and histograms with incompatible custom buckets.
 var invalidCombinationOfHistograms = &histogram.FloatHistogram{}
+
+func IsAggregationDisabled(op parser.ItemType, disabledAggregations []string) bool {
+	for _, aggName := range disabledAggregations {
+		switch strings.ToLower(aggName) {
```

Comment:
Doing `strings.ToLower` on every item in `disabledAggregations` every time we encounter an aggregation seems expensive. What if we converted the list of disabled aggregations to their corresponding `ItemType` in `NewEngine`? This would also give us an opportunity to reject any invalid aggregation names to catch typos etc.

Another optimisation would be to sort the list of disabled `ItemTypes` in `NewEngine` and then do [a binary search](https://pkg.go.dev/slices#BinarySearch) over the list to check if the aggregation is present.